pepper-apis/src/main/java/org/broadinstitute/ddp/service/UserService.java

```diff
@@ -0,0 +1,129 @@
+        handle.attach(JdbiUserStudyEnrollment.class).deleteByUserId(userId);
+        // Removing user addresses
+        handle.attach(JdbiTempMailAddress.class).deleteTempAddressByParticipantId(userId);
+        JdbiMailAddress mailAddressDao = handle.attach(JdbiMailAddress.class);
+        mailAddressDao.deleteDefaultAddressByParticipantId(userId);
+        mailAddressDao.deleteAddressByParticipantId(userId);
+        // Removing all the answers and activity instances
+        ActivityInstanceDao instanceDao = handle.attach(ActivityInstanceDao.class);
+        Set<Long> instanceIds = instanceDao.findAllInstanceIdsByUserIds(Collections.singleton(userId));
+        LOG.info("Found {} activity instances to delete", instanceIds.size());
+        DBUtils.checkDelete(instanceIds.size(), instanceDao.deleteAllByIds(instanceIds));
+        // Removing counters, events
+        handle.attach(JdbiEventConfigurationOccurrenceCounter.class).deleteAllByParticipantId(userId);
+        JdbiQueuedEvent queueEvent = handle.attach(JdbiQueuedEvent.class);
+        queueEvent.deleteQueuedNotificationSubstitutionsByUserIds(Collections.singleton(userId));
+        queueEvent.deleteQueuedNotificationsByUserIds(Collections.singleton(userId));
+        queueEvent.deleteQueuedEventsByUserIds(Collections.singleton(userId));
+        // Removing governances
+        UserGovernanceDao userGovernanceDao = handle.attach(UserGovernanceDao.class);
+        List<Governance> userGovernances = userGovernanceDao.findGovernancesByParticipantGuid(user.getGuid()).collect(Collectors.toList());
+        Set<String> studyGuids = userGovernances.stream()
+                .map(Governance::getGrantedStudies).flatMap(Collection::stream).map(GrantedStudy::getStudyGuid)
+                .collect(Collectors.toSet());
+        userGovernanceDao.deleteAllGovernancesForProxy(userId);
+        userGovernanceDao.deleteAllGovernancesForParticipant(userId);
+        handle.attach(StudyGovernanceSql.class).deleteAgeUpCandidateByParticipantId(userId);
+        // Removing user itself
+        handle.attach(DataExportDao.class).deleteDataSyncRequestsForUser(userId);
+        handle.attach(JdbiUser.class).deleteAllByIds(Collections.singleton(userId));
+
+        // Cleaning up the elasticsearch data
+        RestHighLevelClient esClient = ElasticsearchServiceUtil.getElasticsearchClient(ConfigManager.getInstance().getConfig());
+        BulkRequest bulkRequest = new BulkRequest().timeout("2m");
+        for (String studyGuid : studyGuids) {
+            StudyDto studyDto = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid(studyGuid);
+
+            String indexParticipant = ElasticsearchServiceUtil.getIndexForStudy(handle, studyDto,
+                    ElasticSearchIndexType.PARTICIPANTS);
+            String indexParticipantStructured = ElasticsearchServiceUtil.getIndexForStudy(handle, studyDto,
+                    ElasticSearchIndexType.PARTICIPANTS_STRUCTURED);
```

Comment:
We also need to delete from the `USERS` index for this participant.

Reply:
Added